### PR TITLE
feat(nexus): adding additional statistics for rebuild

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsharma-dc** [comment]
These changes appear to be on top of PR #1293. Are you planning to retain 1293 or, go ahead with all changes on this PR and drop 1293?

**Comment by dsavitskiy** [comment]
@dsharma-dc Normally I would prefer to avoid such situation. But when I need to have a PR on top of another, I first wait until I merge the base PR first, and then I rebase the dependent one and the base commits are gone.
Reviews are supposed to review only the relevant commit in this PR. The base commits are reviewed in a separate PR.

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: io-engine/src/rebuild/rebuild_job_backend.rs [hunk 1]
```diff
@@ -367,33 +370,39 @@ impl RebuildJobBackend {
...
-            segment_size_blks: self.descriptor.segment_size_blks,
+            blocks_per_task: self.descriptor.segment_size_blks,
             block_size: self.descriptor.block_size,
             tasks_total: self.task_pool.total as u64,
             tasks_active: self.task_pool.active as u64,
-        }
+        };
+
+        info!(
+            "State: {state}, Nexus: {nex}, Src: {src}, Dst: {dst}: {res:?}",
```
Comment (tiagolobocastro) [inline io-engine/src/rebuild/rebuild_job_backend.rs:398]: Previously progress was shown, should we keep it as it's kind useful to see it?

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/rebuild/rebuild_job_backend.rs (hunk 1 above)
Comment (dsavitskiy) [reply io-engine/src/rebuild/rebuild_job_backend.rs:398]: It is still shown via {res:?}, res being the stats. Debug output for Stats includes progress.

Comment (tiagolobocastro) [reply io-engine/src/rebuild/rebuild_job_backend.rs:398]: doh, sorry I missed that

**Review by mtzaurus — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors merge

**Review by chriswldenyer — APPROVED** [review]
Looks reasonable

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**